src/main/scala/mesosphere/marathon/state/Migration.scala

```diff
@@ -388,7 +393,36 @@ class MigrationTo1_2(deploymentRepository: DeploymentRepository) {
       await(deploymentRepository.store.expunge(it.next))
     }
 
-    log.info("Finished 1.2 migration")
+    val store = taskRepository.store
+    //scalastyle:off null
+    def loadAndMigrateTasks(id: String): Future[MarathonTaskState] = {
+      store.fetch(id).flatMap {
+        case Some(entity) =>
+          if (entity.toProto.getMarathonTaskStatus == null) {
+            val updatedEntity = entity.toProto.toBuilder
+              .setMarathonTaskStatus(MarathonTaskStatusSerializer.toProto(MarathonTaskStatus(entity.toProto.getStatus)))
+              .build()
+            store.store(id, MarathonTaskState(updatedEntity))
+          } else {
+            Future.successful(entity)
+          }
+        case None => Future.failed(new MigrationFailedException(s"Inconsistency in the task store detected, " +
+          s"task with id $id not found, but delivered in allIds()."))
+      }
+    }
+
+    val migratedTasks = for {
+      ids <- store.names()
+      tasks <- {
+        log.info(s"Discovered ${ids.size} tasks for status migration")
+        Future.sequence(ids.map(loadAndMigrateTasks))
```

Comment:
The ZK stores are unbounded and in the past users have reported issues when we tried to use Future.sequence + ZK operations in migrations.

That's why I remove them serially in lines 391-394. I think we should do something similar here.